### PR TITLE
update php-scoper + bump min PHP version to 7.3

### LINKED ISSUE #488: Support Composer v2
## Feature Request

Currently, phars created by box with Composer 2 are broken.

Example box configuration: https://github.com/jakzal/toolbox/blob/master/box.json.dist

Trying to execute the phar throws a fatal error:

```
PHP Fatal error:  Uncaught Error: Class '_HumbugBoxa72d156a3837\Symfony\Component\Console\Application' not found in phar:///Users/kuba/Projects/toolbox/build/toolbox.phar/src/Cli/Application.php:16
Stack trace:
#0 phar:///Users/kuba/Projects/toolbox/build/toolbox.phar/vendor/composer/ClassLoader.php(444): include()
#1 phar:///Users/kuba/Projects/toolbox/build/toolbox.phar/vendor/composer/ClassLoader.php(322): Composer\Autoload\includeFile('phar:///Users/k...')
#2 [internal function]: Composer\Autoload\ClassLoader->loadClass('_HumbugBoxa72d1...')
#3 phar:///Users/kuba/Projects/toolbox/build/toolbox.phar/bin/toolbox.php(6): spl_autoload_call('_HumbugBoxa72d1...')
#4 /Users/kuba/Projects/toolbox/build/toolbox.phar(12): require('phar:///Users/k...')
#5 {main}
  thrown in phar:///Users/kuba/Projects/toolbox/build/toolbox.phar/src/Cli/Application.php on line 16
```

There are some differences in Composer autoload files between v1 and v2.

Composer 1, `autoload_psr4.php`:

```php
return array(
    '_HumbugBox359b8a3155dc\\Zalas\\Toolbox\\' => array($baseDir . '/src'),
    '_HumbugBox359b8a3155dc\\Symfony\\Polyfill\\Php80\\' => array($vendorDir . '/symfony/polyfill-php80'),
    '_HumbugBox359b8a3155dc\\Symfony\\Polyfill\\Php73\\' => array($vendorDir . '/symfony/polyfill-php73'),
    '_HumbugBox359b8a3155dc\\Symfony\\Polyfill\\Mbstring\\' => array($vendorDir . '/symfony/polyfill-mbstring'),
    '_HumbugBox359b8a3155dc\\Symfony\\Contracts\\Service\\' => array($vendorDir . '/symfony/service-contracts'),
    '_HumbugBox359b8a3155dc\\Symfony\\Component\\Console\\' => array($vendorDir . '/symfony/console'),
    '_HumbugBox359b8a3155dc\\Psr\\Container\\' => array($vendorDir . '/psr/container/src'),
);
```

Composer 2, `autoload_psr4.php`:

```php
return array(
    '_HumbugBoxa72d156a3837\\Zalas\\Toolbox\\' => array($baseDir . '/src'),
    'Symfony\\Polyfill\\Php80\\' => array($vendorDir . '/symfony/polyfill-php80'),
    'Symfony\\Polyfill\\Php73\\' => array($vendorDir . '/symfony/polyfill-php73'),
    'Symfony\\Polyfill\\Mbstring\\' => array($vendorDir . '/symfony/polyfill-mbstring'),
    'Symfony\\Contracts\\Service\\' => array($vendorDir . '/symfony/service-contracts'),
    'Symfony\\Component\\Console\\' => array($vendorDir . '/symfony/console'),
    'Psr\\Container\\' => array($vendorDir . '/psr/container/src'),
);
```

Similar differences can be found in `autoload_static.php`.

### PR DESCRIPTION
**Pull request by smoench**
This will fix also the issue with scoping php-parser in #489

### CODE REVIEW

**Comment by smoench** [comment]
Travis failure is related to https://github.com/nikic/PHP-Parser/issues/701

**Comment by villfa** [comment]
FYI there is also some tests failing because of a breaking change in [Composer 2.0.5](https://github.com/composer/composer/releases/tag/2.0.5) (`platform_check.php` is not generated by default anymore).

**Comment by theofidry** [comment]
Is it something we want now or should I go ahead with another release first?

**Comment by smoench** [comment]
No, not yet. Go ahead with another release :-)

**Comment by smoench** [comment]
There is an issue with scoping php-parser with the current php-scoper version. I could fix it with https://github.com/humbug/php-scoper/pull/437. But I had bump min PHP version to 7.3 as it has been done in php-scoper too.

**Comment by villfa** [comment]
In `Makefile`, I think you should replace `PHPSTAN=bin/phpstan` by `PHPSTAN=vendor-bin/phpstan/vendor/bin/phpstan`.

**Comment by smoench** [comment]
@theofidry Bump min PHP version to 7.3 causes tests failures. Can we revert php-scoper or have a branch with PHP 7.2 support?

**Comment by theofidry** [comment]
@smoench sorry I'm not sure to understand, why do we need 7.2 support?

**Comment by smoench** [comment]
Composer won't let us install the latest php-scoper version as in box's composer.json the platform config for PHP is fixed to 7.2.9.

**Comment by theofidry** [comment]
But if we drop 7.2 (which was the plan) then this value should be bumped right?

**Comment by smoench** [comment]
Docker e2e tests had to be updated. Works now.

**Comment by smoench** [comment]
Waiting for https://github.com/humbug/php-scoper/pull/437 to be merged and a new php-scoper release :)

**Comment by theofidry** [comment]
https://github.com/humbug/php-scoper/releases/tag/0.13.10 released

**Comment by smoench** [comment]
Thank you @theofidry. I've updated the composer.json :-)

File: composer.json [hunk 1]
```diff
@@ -78,7 +78,7 @@
     "config": {
         "bin-dir": "bin",
         "platform": {
-            "php": "7.2.9"
+            "php": "7.3"
```
Comment (theofidry) [inline composer.json:81]: is that correct? I think it needs to be a specific version

**Review by theofidry — APPROVED** [review]
LGTM overall, thanks!

File: composer.json (hunk 1 above)
Comment (smoench) [reply composer.json:81]: Composer doesn't complain about it:
```
$ composer validate
./composer.json is valid
```

Should I adjust it to `7.3.0`?

Comment (theofidry) [reply composer.json:81]: No it's more convenient if that just works!

**Pull request merged**